misc/efi-stub/Makefile

```diff
@@ -0,0 +1,115 @@
+#
+# Copyright (c) 2011 - 2021, Intel Corporation
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#    * Redistributions of source code must retain the above copyright
+#      notice, this list of conditions and the following disclaimer.
+#    * Redistributions in binary form must reproduce the above
+#      copyright notice, this list of conditions and the following
+#      disclaimer in the documentation and/or other materials provided
+#      with the distribution.
+#    * Neither the name of Intel Corporation nor the names of its
+#      contributors may be used to endorse or promote products derived
+#      from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+
+HV_OBJDIR:=build
+HV_SRC:=../../hypervisor
+C_SRCS = boot.c pe.c malloc.c container.c
+ACRN_OBJS := $(patsubst %.c,$(EFI_OBJDIR)/%.o,$(C_SRCS))
+INCLUDE_PATH += $(INCDIR)/efi
+INCLUDE_PATH += $(INCDIR)/efi/$(ARCH)
+INCLUDE_PATH += $(HV_OBJDIR)/include
+INCLUDE_PATH += $(HV_SRC)/include/arch/x86
+INCLUDE_PATH += $(HV_SRC)/include/arch/x86/guest
+INCLUDE_PATH += $(HV_SRC)/include/public
+INCLUDE_PATH += $(HV_SRC)/include/lib
+INCLUDE_PATH += $(HV_SRC)/boot/include/guest
+
+BOARD ?= tgl-rvp
+SCENARIO ?= hybrid_rt
+
+OBJCOPY=objcopy
+
+HOST  = $(shell $(CC) -dumpmachine | sed "s/\(-\).*$$//")
+ARCH := $(shell $(CC) -dumpmachine | sed "s/\(-\).*$$//")
+
+ifeq ($(ARCH),x86_64)
+	FORMAT=efi-app-x86-64
```

Comment:
A couple of questions regarding this logic:
* If I understand it correctly, we are essentially deriving the target architecture format based on the output from `gcc -dumpmachine`. I believe our objective here is to actually generate a 32-bit EFI app or 64-bit EFI app based on whether the EFI firmware is itself 32-bit or 64-bit. Is it guaranteed that `gcc -dumpmachine` output will match the EFI firmward format? Or is it possible that it returns that we're running a 32-bit OS on a 64-bit EFI firmward (or vice-versa)?
* Regardless of the answer above, if we assume that the user may be building on a dev system which is different than the target system, it's likely better to not 'guess' the architecture but rather set it via a compile-time variable.

Thoughts?

Reply:
Thank you. I will remove the if condition. The expectation is to build a 64-bit EFI app.

Reply:
Thanks for the confirmation.